src/plugins_types.c

```diff
@@ -506,33 +505,34 @@ ly_type_validate_patterns(struct lysc_pattern **patterns, const char *str, size_
         }
 
         rc = pcre2_match(patterns[u]->code, (PCRE2_SPTR)str, str_len, 0, PCRE2_ANCHORED | PCRE2_ENDANCHORED, match_data, NULL);
-        if (rc == PCRE2_ERROR_NOMATCH) {
-            if (asprintf(&errmsg, "String \"%.*s\" does not conform to the pattern \"%s\".", (int)str_len, str, patterns[u]->expr) == -1) {
+        pcre2_match_data_free(match_data);
+
+        /* ERROR CHECK */
+        /* !(rc == PCRE2_ERROR_NOMATCH || rc >= 0) */
```

Comment:
Why don't you reverse the if's branches? Start with a positive result/branch.

Reply:
Because the next condition (rc != PCRE2_ERROR_NOMATCH) && (patterns[u]->inverted == 1) would have to be little more complicated in that case. It would have to be changed to  (rc >= 0) && (patterns[u]->inverted == 1)